tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -403,3 +459,22 @@ ROOT::Experimental::RResult<void> ROOT::Experimental::RNTupleImporter::Import()
 
    return RResult<void>::Success();
 }
+
+ROOT::Experimental::RResult<void> ROOT::Experimental::RNTupleImporter::SetNTupleName(const std::string &ntupleName)
+{
+   if (fSourceTrees.size() > 1) {
+      return R__FAIL("Multiple trees are being imported, please specify the source tree name");
+   }
+   fNTupleNames.at(fSourceTrees.at(0)->GetName()) = ntupleName;
+   return RResult<void>::Success();
+}
+
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::RNTupleImporter::SetNTupleName(std::string_view treeName, const std::string &ntupleName)
+{
+   if (fNTupleNames.find(treeName) == fNTupleNames.end()) {
+      return R__FAIL("Tree '" + std::string(treeName) + "' not present in " + std::string(fSourceFile->GetName()));
+   }
+   fNTupleNames.at(treeName) = ntupleName;
```

Comment:
We might totally avoid a second lookup in the map (caused by the `at()` call) if we store the iterator returned by `find()`.  Dereferencing such iterator should result in an lvalue that may be assigned to, as in
```suggestion
   auto it = fNTupleNames.find(treeName);
   if (it == fNTupleNames.end())
      return R__FAIL("Tree '" + std::string(treeName) + "' not present in " + std::string(fSourceFile->GetName()));
   *it = ntupleName;
```